<!-- MHonArc v2.6.10 -->
<!--X-Subject: RE: [bug #7755] modsel.py error with "eliminate" &#45; still happening! -->
<!--X-From-R13: "Vrnqrl, Egrcura" <E.X.VrnqrlNznffrl.np.am> -->
<!--X-Date: Mon, 27 Nov 2006 05:10:21 +0100 -->
<!--X-Message-Id: 59ADAF5CA6A8FC4C889BD4097D912BB1072D3330@its&#45;xchg4.massey.ac.nz -->
<!--X-Content-Type: text/plain -->
<!--X-Reference: 20061117&#45;050140.sv6928.43673@gna.org -->
<!--X-Reference: 20061120&#45;214715.sv6928.33029@gna.org -->
<!--X-Reference: 20061121&#45;163826.sv2782.17895@gna.org -->
<!--X-Reference: 59ADAF5CA6A8FC4C889BD4097D912BB1072D332D@its&#45;xchg4.massey.ac.nz -->
<!--X-Reference: 7f080ed10611240029h624b3ec7k43a45649b0aeae9@mail.gmail.com -->
<!--X-Head-End-->
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
   "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
<title>RE: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening! -- November 27, 2006 - 05:10</title>
<link rel="stylesheet" type="text/css" href="/mail.gna.org/archives-color-gna.css"> 
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
</head>
<body>
<!--X-Body-Begin-->
<!--X-User-Header-->
<!--X-User-Header-End-->
<!--X-TopPNI-->
<h2><img src="/mail.gna.org/images/mail.orig.png" width="48" height="48"
alt="mail" class="pageicon" />RE: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening!</h2>
<br />
<div class="topmenu">
<a href="../" class="tabs">Others Months</a> | <a href="index.html#00205" class="tabs">Index by Date</a> | <a href="threads.html#00205" class="tabs">Thread Index</a><br />
<span class="smaller">&gt;&gt;&nbsp;&nbsp;
[<a href="msg00204.html">Date Prev</a>] [<a href="msg00206.html">Date Next</a>] [<a href="msg00201.html">Thread Prev</a>] [<a href="msg00208.html">Thread Next</a>]
</div>

<!--X-TopPNI-End-->
<!--X-MsgBody-->
<!--X-Subject-Header-Begin-->
<h3><a name="header" href="#header">Header</a></h3>
<!--X-Subject-Header-End-->
<!--X-Head-of-Message-->
<ul class="headdata">
<li class="menuitem">
<em>To</em>: &quot;Edward d'Auvergne&quot; &lt;edward.dauvergne@xxxxxxxxx&gt;</li>
<li class="menuitem">
<em>Date</em>: Mon, 27 Nov 2006 17:08:45 +1300</li>
<li class="menuitem">
<em>Cc</em>: Edward d Auvergne &lt;NO-REPLY.INVALID-ADDRESS@xxxxxxx&gt;, relax-devel@xxxxxxx</li>
<li class="menuitem">
<em>Message-id</em>: &lt;<a href="msg00205.html">59ADAF5CA6A8FC4C889BD4097D912BB1072D3330@its-xchg4.massey.ac.nz</a>&gt;</li>
<li class="menuitem">
<em>References</em>: &lt;<a href="msg00164.html">20061117-050140.sv6928.43673@gna.org</a>&gt;&lt;<a href="msg00178.html">20061120-214715.sv6928.33029@gna.org</a>&gt;&lt;<a href="msg00188.html">20061121-163826.sv2782.17895@gna.org</a>&gt;&lt;<a href="msg00200.html">59ADAF5CA6A8FC4C889BD4097D912BB1072D332D@its-xchg4.massey.ac.nz</a>&gt;	&lt;<a href="msg00201.html">7f080ed10611240029h624b3ec7k43a45649b0aeae9@mail.gmail.com</a>&gt;</li>
<li class="menuitem">
<em>Thread-index</em>: AccPop8p4Fm7YbvSQYm67Z1NpnNpAwCNu0g8</li>
<li class="menuitem">
<em>Thread-topic</em>: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening!</li>
</ul>
<!--X-Head-of-Message-End-->
<!--X-Head-Body-Sep-Begin-->
</div><!-- end headdata -->
<br />
<h3><a name="content" href="#content">Content</a></h3>
<div class="postedby">Posted by <strong>Headey, Stephen</strong> on November 27, 2006 - 05:10:</div>
<div class="msgdata">
<!--X-Head-Body-Sep-End-->
<!--X-Body-of-Message-->
<pre style="margin: 0em;">Hi Edward,

I got the same error message for model 5 when I ran 300 simulations, so there 
must be some other factor at play, unless 300 is also not enough.

Cheers,
Steve


-----Original Message-----
From:   Edward d'Auvergne [<a  href="mailto:edward.dauvergne@xxxxxxxxx">mailto:edward.dauvergne@xxxxxxxxx</a>]
Sent:   Fri 11/24/2006 9:29 PM
To:     Headey, Stephen
Cc:     Edward d Auvergne; relax-devel@xxxxxxx
Subject:        Re: [bug #7755] modsel.py error with &quot;eliminate&quot; - still 
happening!

Yes, I have replicated and confirmed bug #7755
(<a  href="http://gna.org/bugs/?7755">http://gna.org/bugs/?7755</a>).  I'm in the process of working out a
solution.  The ZeroDivisionError is another bug I have encountered
whereby all Monte Carlo simulations have failed and have been
eliminated.  Hence n, the number of Monte Carlo simulations is zero
and the error occurs.  The problem here is that not enough MC
simulations have been run, but I'll fix the error message anyway.

Edward


On 11/24/06, Headey, Stephen &lt;S.J.Headey@xxxxxxxxxxxx&gt; wrote:
</pre><blockquote class="blockquote"><pre style="margin: 0em;">Hi Edward,

I've sorted out the corrupt data files and started again with 
model-free.py. (New data files attached to bug #7755). Models 1-4 seem to 
run OK. Model 5 gives the following error after MC simulations.

relax&gt; minimise(*args=('newton',), run='m5', func_tol=1e-25, 
max_iterations=10000000, constraints=1, scaling=1, print_flag=1)
Simulation 1
Simulation 2
Simulation 3
Simulation 4
Simulation 5
Simulation 6
Simulation 7
Simulation 8
Simulation 9
Simulation 10

relax&gt; eliminate(run='m5', function=None, args=None)

relax&gt; monte_carlo.error_analysis(run='m5', prune=0.0)
Traceback (most recent call last):
  File &quot;/usr/local/bin/relax&quot;, line 454, in ?
    Relax()
  File &quot;/usr/local/bin/relax&quot;, line 166, in __init__
    self.interpreter.run()
  File &quot;/usr/local/relax/prompt/interpreter.py&quot;, line 213, in run
    run_script(intro=self.relax.intro_string, local=self.local, 
script_file=self.relax.script_file, quit=1)
  File &quot;/usr/local/relax/prompt/interpreter.py&quot;, line 388, in run_script
    console.interact(intro, local, script_file, quit)
  File &quot;/usr/local/relax/prompt/interpreter.py&quot;, line 340, in 
interact_script
    execfile(script_file, local)
  File &quot;model-free2.py&quot;, line 57, in ?
    monte_carlo.error_analysis(name)
  File &quot;/usr/local/relax/prompt/monte_carlo.py&quot;, line 145, in error_analysis
    self.__relax__.generic.monte_carlo.error_analysis(run=run, prune=prune)
  File &quot;/usr/local/relax/generic_fns/monte_carlo.py&quot;, line 219, in 
error_analysis
    Xav = Xsum / float(n)
ZeroDivisionError: float division


I also ran the modsel.py script specifying models 1-4. I still get the same 
&quot;list index out of range&quot; error.

relax&gt; eliminate(run=None, function=None, args=None)
Traceback (most recent call last):
  File &quot;/usr/local/software/relax-1.2/relax&quot;, line 454, in ?
    Relax()
  File &quot;/usr/local/software/relax-1.2/relax&quot;, line 166, in __init__
    self.interpreter.run()
  File &quot;/usr/local/software/relax-1.2/prompt/interpreter.py&quot;, line 213, in 
run
    run_script(intro=self.relax.intro_string, local=self.local, 
script_file=self.relax.script_file, quit=1)
  File &quot;/usr/local/software/relax-1.2/prompt/interpreter.py&quot;, line 388, in 
run_script
    console.interact(intro, local, script_file, quit)
  File &quot;/usr/local/software/relax-1.2/prompt/interpreter.py&quot;, line 340, in 
interact_script
    execfile(script_file, local)
  File &quot;modsel.py&quot;, line 20, in ?
    eliminate()
  File &quot;/usr/local/software/relax-1.2/prompt/eliminate.py&quot;, line 97, in 
eliminate
    self.relax.generic.eliminate.eliminate(run=run, function=function, 
args=args)
  File &quot;/usr/local/software/relax-1.2/generic_fns/eliminate.py&quot;, line 100, 
in eliminate
    values = param_values(self.run, i, sim_index=j)
  File &quot;/usr/local/software/relax-1.2/specific_fns/model_free.py&quot;, line 
1499, in get_param_values
    self.param_vector = self.assemble_param_vector(index=index, 
sim_index=sim_index)
  File &quot;/usr/local/software/relax-1.2/specific_fns/model_free.py&quot;, line 
223, in assemble_param_vector
    param_vector.append(self.relax.data.res[self.run][i].s2_sim[sim_index])
IndexError: list index out of range

Cheers,
Stephen

-----Original Message-----
From:   Edward d Auvergne [<a  href="mailto:NO-REPLY.INVALID-ADDRESS@xxxxxxx">mailto:NO-REPLY.INVALID-ADDRESS@xxxxxxx</a>]
Sent:   Wed 11/22/2006 5:38 AM
To:     Edward d Auvergne; Headey, Stephen; relax-devel@xxxxxxx
Cc:
Subject:        [bug #7755] modsel.py error with &quot;eliminate&quot;


Update of bug #7755 (project relax):

                  Status:                    None =&gt; Confirmed
        Operating System:               GNU/Linux =&gt; All systems

    _______________________________________________________

Follow-up Comment #1:

The problem here is that the Monte Carlo simulation data structures are not
being properly recreated by the model-free results reading functions.  I
tried this using 10 Monte Carlo simultations together with the relaxation
data attached to bug #7787 (<a  href="https://gna.org/bugs/?7787">https://gna.org/bugs/?7787</a>).  After reading the
results files, self.relax.data.sim_number[run] was set to 1300!  This is 10
times 130 spins!  Hence elements beyond the end of the Monte Carlo 
simulation
arrays (of length 10) are being accessed.

    _______________________________________________________

Reply to this item at:

  &lt;<a  href="http://gna.org/bugs/?7755">http://gna.org/bugs/?7755</a>&gt;

_______________________________________________
  Message sent via/by Gna!
  <a  href="http://gna.org/">http://gna.org/</a>





</pre></blockquote><pre style="margin: 0em;">





</pre>
<!--X-Body-of-Message-End-->
<!--X-MsgBody-End-->
<!--X-Follow-Ups-->
</div><!-- end msgdata -->
<br />
<h3><a name="related" href="#related">Related Messages</a></h3>
<div class="relateddata">
<ul><li><strong>Follow-Ups</strong>:
<ul>
<li><strong><a name="00208" href="msg00208.html">Re: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening!</a></strong>
<ul><li><em>From:</em> Edward d'Auvergne</li></ul></li>
</ul></li></ul>
<!--X-Follow-Ups-End-->
<!--X-References-->
<ul><li><strong>References</strong>:
<ul>
<li><strong><a name="00164" href="msg00164.html">[bug #7755] modsel.py error with &quot;eliminate&quot;</a></strong>
<ul><li><em>From:</em> Stephen Headey</li></ul></li>
<li><strong><a name="00178" href="msg00178.html">[bug #7755] modsel.py error with &quot;eliminate&quot;</a></strong>
<ul><li><em>From:</em> Stephen Headey</li></ul></li>
<li><strong><a name="00188" href="msg00188.html">[bug #7755] modsel.py error with &quot;eliminate&quot;</a></strong>
<ul><li><em>From:</em> Edward d Auvergne</li></ul></li>
<li><strong><a name="00200" href="msg00200.html">RE: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening!</a></strong>
<ul><li><em>From:</em> Headey, Stephen</li></ul></li>
<li><strong><a name="00201" href="msg00201.html">Re: [bug #7755] modsel.py error with &quot;eliminate&quot; - still happening!</a></strong>
<ul><li><em>From:</em> Edward d'Auvergne</li></ul></li>
</ul></li></ul>
<!--X-References-End-->
<!--X-BotPNI-->
</div><!-- end relateddata -->
<!-- NoBotLinksApartFromRelatedMessages -->

<!--X-BotPNI-End-->
<!--X-User-Footer-->
<!--X-User-Footer-End-->
<div class="footer"></div><br />
<div class="right">Powered by <a href="http://www.mhonarc.org">MHonArc</a>, Updated Tue Nov 28 04:40:50 2006</div>  
</body>
</html>
